cmd/admin/key.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os/user"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/privatekey"
+	sapb "github.com/letsencrypt/boulder/sa/proto"
+)
+
+// subcommandBlockKey encapsulates the "admin block-key" command.
+func (a *admin) subcommandBlockKey(ctx context.Context, args []string) error {
+	subflags := flag.NewFlagSet("block-key", flag.ExitOnError)
+	privKey := subflags.String("private-key", "", "Block issuance for the pubkey corresponding to this private key")
+	comment := subflags.String("comment", "", "Additional context to add to database comment column")
+	_ = subflags.Parse(args)
+
+	if *privKey == "" {
+		return errors.New("the -private-key flag is required")
+	}
+
+	spkiHash, err := a.spkiHashFromPrivateKey(*privKey)
+	if err != nil {
+		return err
+	}
+
+	err = a.blockSPKIHash(ctx, spkiHash, *comment)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *admin) spkiHashFromPrivateKey(keyFile string) ([]byte, error) {
+	_, publicKey, err := privatekey.Load(keyFile)
+	if err != nil {
+		return nil, fmt.Errorf("loading private key file: %w", err)
+	}
+
+	spkiHash, err := core.KeyDigest(publicKey)
+	if err != nil {
+		return nil, fmt.Errorf("computing SPKI hash: %w", err)
+	}
+
+	return spkiHash[:], nil
+}
+
+func (a *admin) blockSPKIHash(ctx context.Context, spkiHash []byte, comment string) error {
+	var count int
```

Comment:
Should there be a [query](https://github.com/letsencrypt/boulder/blob/2f114f8a9b9612a6d57774cab40e3620e3ec4533/cmd/admin-revoker/main.go#L409-L420) to determine if the SPKI is already in the blockedKeys table and avoid doing the work below?

Reply:
Done!